packages/scripts/src/codegen/icon-codegen.ts

```diff
@@ -0,0 +1,90 @@
+import * as path from 'path'
+import * as fs from 'fs'
+import { src, watch } from 'gulp'
+import { prettier, ROOT_PATH, watchTask } from '../utils'
+import glob from 'glob-promise'
+import { camelCase } from 'lodash-unified'
+
+const pattern = 'packages/icons/**/*.@(svg|jpe?g|png)'
+const iconRoot = path.resolve(__dirname, '../../../icons')
+const CODE_FILE = path.resolve(iconRoot, 'icon-data.ts')
+
+const currentColorRe = /\w=('|")currentColor\1/
+
+const hasCurrentColor = (code: string) => currentColorRe.test(code)
+
+function optimizeSvg(code: string) {
+    return (
+        code
+            .trim()
+            // set both height and width to 100%, let svg's container, aka <Icon />, decide the size
+            // We don't use g flag here, because we only want to change the first attribute of each
+            .replace(/\b(height)=('|")\d+\2/, '$1="100%"')
+            .replace(/\b(width)=('|")\d+\2/, '$1="100%"')
+    )
+}
+
+export async function generateIcons() {
+    const nameMap: Record<string, string> = {}
+    const iconsWithDynamicColor: string[] = []
+    const lines: string[] = []
+    const names: string[] = []
+
+    const filePaths = await glob.promise(pattern, { cwd: ROOT_PATH, nodir: true })
+    filePaths.forEach((filePath) => {
+        const parsed = path.parse(filePath)
+        const fileName = parsed.base
+        const name = camelCase(parsed.name)
+        if (name.toLowerCase() === 'redpacket') {
+            return
+        }
+        names.push(name)
```

Comment:
Please warn or fail if the icon name already appeared (which means there is name conflict).

Reply:
Will do.


```ts
        if (name.toLowerCase() === 'redpacket') {
            return
        }
```

To clarify, by the way, this is workaround for POC.